scripts/commands/addlights.lua

```diff
@@ -0,0 +1,60 @@
+---------------------------------------------------------------------------------------------------
+-- func: addlights <light type> <amount> <target player>
+-- desc: Adds the amount of specified light to the player
+---------------------------------------------------------------------------------------------------
+-- lights: 1: pearl, 2: azure, 3: ruby, 4: amber, 5: gold, 6: silver, 7: ebon
+---------------------------------------------------------------------------------------------------
+require("scripts/globals/abyssea")
+
+cmdprops =
+{
+    permission = 1,
+    parameters = "sis"
+}
+
+local lightType =
+{
+    pearl  = 1,
+    azure  = 2,
+    ruby   = 3,
+    amber  = 4,
+    gold   = 5,
+    silver = 6,
+    ebon   = 7,
+}
+
+function error(player, msg)
+    player:PrintToPlayer(msg)
+    player:PrintToPlayer("!addlights <light type> <amount> {player}")
+end
+
+function onTrigger(player,light,amount,target)
+    -- validate target
+    local targ
+    if (target == nil) then
+        targ = player
+    else
+        targ = GetPlayerByName(target)
+        if (targ == nil) then
+            error(player, string.format("Player named '%s' not found!", target))
+            return
+        end
+    end
+
+    local selectedLight = tostring(light)
+    local setLight = lightType[selectedLight]
+    if (light == nil or setLight == nil) then
```

Comment:
I think you will get a lua "nil index" error if selectedLight is nil (you may want to bomb out separately before line 45)

Reply:
yep, good call, cheers whasf

Reply:
changed